feature-extensions/splunk/setup.py

```diff
@@ -0,0 +1,142 @@
+        help='Specify the splunk index'
+    )
+    mainParser.add_argument(
+        '--scm-repo',
+        help='Specify the scm repo url'
+    )
+    mainParser.add_argument(
+        '--scm-username',
+        help='Specify the scm username'
+    )
+    mainParser.add_argument(
+        '--scm-password',
+        help='Specify the scm password'
+    )
+    mainParser.add_argument(
+        '--scm-pathext',
+        help='Specify the scm repo path ext (Use "scm" for bitbucket)'
+    )
+
+    args = mainParser.parse_args()
+    args.func(args)
+
+
+def install(args):
+    print(
+        colors.OKGREEN +
+        "\nThis will install {0} functionality into your Jazz deployment.\n".format(featureName)
+        + colors.ENDC)
+
+    configureSplunk(args, True)
+
+
+def configureSplunk(args, splunk_enable):
+
+    if not args.splunk_endpoint:
+        args.splunk_endpoint = raw_input("Please enter the Splunk Endpoint: ")
+
+    if not args.splunk_token:
+        args.splunk_token = raw_input("Please enter the Splunk Token: ")
+
+    if not args.splunk_index:
+        args.splunk_index = raw_input("Please enter the Splunk Index: ")
+
+    if not args.scm_repo:
+        args.scm_repo = raw_input("Please enter the SCM Repo: ")
+
+    if not args.scm_username:
+        args.scm_username = raw_input("Please enter the SCM Username: ")
+
+    if not args.scm_password:
+        args.scm_password = raw_input("Please enter the SCM Password: ")
+
+    if not args.scm_pathext:
+        args.scm_pathext = raw_input("Please enter the Splunk Pathext (Use \"/scm\" for bitbucket): ") or "/"
+
+    if not args.splunk_endpoint and not args.splunk_token and not args.splunk_index:
+        print(colors.FAIL +
+              'Cannot proceed! No install possible'
+              + colors.ENDC)
+        return True
+
+    replace_config(args)
+
+
+def replace_config(args):
+
+    # Clone the SCM
+    subprocess.check_call(
+        [
+            "git",
+            "clone",
+            ("http://%s:%s@%s%s/slf/jazz-build-module.git") %
+            (args.scm_username,
+             urllib.quote(
+                 args.scm_password),
+             args.scm_repo,
+             args.scm_pathext),
+            "--depth",
+            "1"])
+
+    configFile = "jazz-installer-vars.json"
+    # Read in the file
+    with open('./jazz-build-module/'+configFile, 'r') as file:
+        filedata = file.read()
+
+    # Replace the target string
+    filedata = filedata.replace('"{ENABLE_SPLUNK_FEATURE}"', 'true')
+    filedata = filedata.replace('{SPLUNK_ENDPOINT}', args.splunk_endpoint)
+    filedata = filedata.replace('{SPLUNK_TOKEN}', args.splunk_token)
+    filedata = filedata.replace('{SPLUNK_INDEX}', args.splunk_index)
+
+    # Write the file out again
+    with open('./jazz-build-module/'+configFile, 'w') as file:
+        file.write(filedata)
+
+    # Commit the changes
+    subprocess.check_call(["git", "add", configFile], cwd='./jazz-build-module')
```

Comment:
Since we're using the string `jazz-build-module` in a bunch of spots, can you please pull it out into a toplevel variable?

Reply:
Done